src/main/java/io/hgraphdb/models/EdgeIndexModel.java

```diff
@@ -163,37 +165,31 @@ public void deleteEdgeIndex(Edge edge, String key, Long ts) {
     @SuppressWarnings("unchecked")
     private Iterator<Edge> performEdgesScan(HBaseVertex vertex, Scan scan, Tuple cacheKey,
                                             boolean useIndex, Predicate<HBaseEdge> filter) {
-        List<Edge> cached = new ArrayList<>();
         final EdgeIndexReader parser = new EdgeIndexReader(graph);
         ResultScanner scanner;
         try {
             scanner = table.getScanner(scan);
-            Iterator<Edge> iterator = CloseableIteratorUtils.flatMap(
-                    CloseableIteratorUtils.concat(scanner.iterator(), IteratorUtils.of(Result.EMPTY_RESULT)),
+            List<Edge> edges = Lists.newArrayList(scanner.iterator())
```

Comment:
Please check if `graph.isParallelLoading()` and then retain the old code if `false`.